app/server/inspector/src/main/java/io/syndesis/server/inspector/JsonInstanceInspector.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.syndesis.server.inspector;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import io.syndesis.common.util.Json;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Christoph Deppisch
+ */
+@Component
+public class JsonInstanceInspector implements Inspector {
+
+    private static final Logger LOG = LoggerFactory.getLogger(JsonInstanceInspector.class);
+
+    private static final String ARRAY_CONTEXT = "[]";
+    static final List<String> COLLECTION_PATHS = Collections.singletonList("size()");
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<String> getPaths(String kind, String type, String specification, Optional<byte[]> exemplar) {
+        if (specification == null) {
```

Comment:
Should we test on `"json-instance".equals(kind)`?

Reply:
Perhaps also in the `JsonSchemaInspector`

Reply:
This check on the kind is already done in the `supports` method which is called when evaluating matching inspectors